org-static-blog.el

```diff
@@ -265,6 +266,36 @@ Only if og tags are enabled. It can be overridden with the
   :type '(string)
   :safe t)
 
+(defcustom org-static-blog-use-semantic-html nil
+  "Use semantic html tags instead of <div>.
+
+Semantic html output is customizable via `org-static-blog-semantic-html-mapping'"
+  :type '(boolean)
+  :safe t)
+
+(defvar org-static-blog-semantic-html-mapping
+  '((preamble "header" "id=\"preamble\" class=\"status\"")
```

Comment:
Is that mapping correct? We do have a page header and footer for each page. The preamble and postamble are part of the content really, not of the page.

Reply:
I was mostly following your instructions here: https://github.com/bastibe/org-static-blog/issues/8#issuecomment-383008110

> The preamble and postamble are part of the content really, not of the page.

Mh, I 'm not sure if I understand what you mean by the difference between content and page.
Isn't the preamble and postamble the same for all pages?

https://github.com/bastibe/org-static-blog/blob/682dbb6f78926cdea332e91e9e9ff40bc68d1351/org-static-blog.el#L159-L167

The structure in the `org-static-blog-template` is
- preamble
- content
- postamble

https://github.com/bastibe/org-static-blog/blob/682dbb6f78926cdea332e91e9e9ff40bc68d1351/org-static-blog.el#L396-L405

`tContent` is either a single post or a multipost-page.

From https://html.spec.whatwg.org/multipage/sections.html#the-header-element:

> The [header](https://html.spec.whatwg.org/multipage/sections.html#the-header-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) a group of introductory or navigational aids.

This is how the preamble is currently used in the blogs linked in the README.

Also see the last `<header>` example from the spec linked above:

```html
<body>
 <header>
  <h1>Little Green Guys With Guns</h1>
  <nav>
   <ul>
    <li><a href="/games">Games</a>
    <li><a href="/forum">Forum</a>
    <li><a href="/download">Download</a>
   </ul>
  </nav>
  <h2>Important News</h2> <!-- this starts a second subsection -->
  <!-- this is part of the subsection entitled "Important News" -->
  <p>To play today's games you will need to update your client.</p>
  <h2>Games</h2> <!-- this starts a third subsection -->
 </header>
 <p>You have three active games:</p>
 <!-- this is still part of the subsection entitled "Games" -->
 ...
```

From the `<footer>` [spec](https://html.spec.whatwg.org/multipage/sections.html#the-footer-element):

> The [footer](https://html.spec.whatwg.org/multipage/sections.html#the-footer-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) a footer for its nearest ancestor [sectioning content](https://html.spec.whatwg.org/multipage/dom.html#sectioning-content-2) element, or for [the body element](https://html.spec.whatwg.org/multipage/dom.html#the-body-element-2) if there is no such ancestor. A footer typically contains information about its section such as who wrote it, links to related documents, copyright data, and the like.
> 
> When the [footer](https://html.spec.whatwg.org/multipage/sections.html#the-footer-element) element contains entire sections, they [represent](https://html.spec.whatwg.org/multipage/dom.html#represents) appendices, indices, long colophons, verbose license agreements, and other such content.

This is also the usage that I found in the blogs that I had a look at.